### ISSUE TITLE
Unicode in REPL on Windows 

### ISSUE THREAD

**Issue by ForNeVeR** [morganey-lang/Morganey#95]
It'll be a tough problem, but we should take a look.

So, currently Morganey REPL uses Unicode characters such as `λ` in its prompt, but nobody was able to see them in Windows terminal. Ordinary tricks such as `chcp 65001` weren't working for me.

We should research the issue and maybe invent some workaround.

**Comment by ghost** [comment]
change font tu lucida console

**Comment by ForNeVeR** [comment]
I am using unicode font (not Lucida Console, but [Liberatior Mono Powerline](https://github.com/powerline/fonts/tree/master/LiberationMono)) and I can see the `λ` character in my terminal without problems. It's definitely Morganey (or maybe JVM) fault. Investigating right now.

**Comment by ForNeVeR** [comment]
Here's the proof:
![image](https://cloud.githubusercontent.com/assets/92793/18054259/4b3833d4-6e2d-11e6-84d3-a6d97b188778.png)

**Comment by ForNeVeR** [comment]
Well, it seems that it's a fundamental problem of HotSpot implementation on Windows. No matter what (`-Dfile.encoding=UTF-8` or creating various kinds of UTF-8 output streams) it will use the **wrong**, single byte API on Windows (when it should really use text API aka `WriteConsoleW`). We could cast some magicks involving UTF-8, `chcp` etc, but I'm not so interested in that. Too trivial and too error-prone at the same time. I have a better idea and have planned to conduct some utilitary research in the coming times. The task will be solved in an excellent way.

**Comment by ghost** [comment]
I found this interesting thing we should try it: http://stackoverflow.com/questions/5965195/utf-8-cjk-characters-not-displaying-in-java. First answer

**Comment by ForNeVeR** [comment]
It _sort of_ works if I use UTF-8 and `chcp 65001`. I've added the following into the main method:

``` scala
  def main(args: Array[String]) = {
    val stdout = new java.io.PrintStream(System.out, true, "UTF-8")
    stdout.println("λλλ")
    ...
  }
```

And I got the following:

![image](https://cloud.githubusercontent.com/assets/92793/18081420/51b5219e-6ec4-11e6-9bc1-c515c8afd3c2.png)

No idea why is it printing `λλλ�λ�` instead of `λλλ`; without `chcp 65001` it outputs the following: `╬╗╬╗╬╗`. And I cannot enter Cyrillic characters in this mode (Windows terminal sometimes becomes broken after changing the encodings this way).

As I said, this way is too error-prone and it's simply not right: we're trying to mimic text output using **byte**-level APIs (as defined by HotSpot). We shouldn't follow this way.

**Comment by keddelzz** [comment]
As long this issue is not resolved I propose implementing a workaround, which prints `\` instead of `λ` (on windows only). This makes the result of an evaluation copy-pasteable as the input of another evaluation.

**Comment by rexim** [comment]
Yeah, right now the Morganey REPL on Windows looks like a Prolog prompt :D

**Comment by rexim** [comment]
@ForNeVeR I know how you like Prolog! ;)

**Comment by ForNeVeR** [comment]
_(Moved from https://github.com/morganey-lang/Morganey/issues/196#issuecomment-265059969.)_

You know guys what's interesting? I'm running [my another project](https://github.com/ForNeVeR/checktimer) in Windows 8.1 terminal now, and it seems that it handles Unicode properly if executing _without_ sbt (i.e. `java -cp .../scala-xxx.jar;./checktimer.jar my.fornever.Class`), but it **cannot** handle the Unicode characters if executing from sbt.

Needs additional experimentation.

**Comment by ForNeVeR** [comment]
Nope, it doesn't work with Morganey.

**Comment by rexim** [comment]
@ForNeVeR as far as I know both sbt and Morganey use JLine.

**Comment by ForNeVeR** [comment]
Okay, I'll test with pure Java application. Probably that's really a JLine issue, thanks for the info.

**Comment by ForNeVeR** [comment]
Nope, it doesn't work in pure Java application. Definitely a JVM issue.
